docs/copied-from-beats/docs/security/api-keys.asciidoc

```diff
@@ -19,71 +19,114 @@ IMPORTANT: Review <<feature-roles>> before creating API keys for {beatname_uc}.
+For example:
+
+[source,json,subs="attributes,callouts"]
+----
 {
-  "name": "{beat_default_index_prefix}_host001", <1>
-  "role_descriptors": {
-    "{beat_default_index_prefix}_writer": { <2>
-      "cluster": ["monitor", "read_ilm"],
-      "index": [
-        {
-          "names": ["{beat_default_index_prefix}-*"],
-          "privileges": ["view_index_metadata", "create_doc"]
-        }
+    "{beat_default_index_prefix}_writer": {
+        "cluster": ["monitor", "read_ilm"], <1>
+        "index": [ <2>
+            {
+                "names": ["{beat_default_index_prefix}-*"],
+                "privileges": ["view_index_metadata", "create_doc"]
+            }
       ]
     }
-  }
 }
-------------------------------------------------------------
-<1> Name of the API key
-<2> Granted privileges, see <<feature-roles>>
+----
+<1> Defines cluster-level privileges
+<2> Defines index-level privileges
 
 NOTE: See <<privileges-to-publish-events>> for the list of privileges required to publish events.
 
-The return value will look something like this:
+To set an expiration date for the API key, select **Expire after time**
+and input the lifetime of the API key in days.
 
-[source,console-result,subs="attributes,callouts"]
---------------------------------------------------
-{
-  "id":"TiNAGG4BaaMdaH1tRfuU", <1>
-  "name":"{beat_default_index_prefix}_host001",
-  "api_key":"KnR6yE41RrSowb0kQ0HWoA" <2>
-}
---------------------------------------------------
-<1> Unique id for this API key
-<2> Generated API key
+Click **Create API key**. In the dropdown, switch to **Beats* and copy the API key.
 
-You can now use this API key in your +{beatname_lc}.yml+ configuration file like this:
-["source","yaml"]
+You can now use this API key in your +{beatname_lc}.yml+ configuration file:
+
+["source","yml",subs="attributes"]
 --------------------
 output.elasticsearch:
   api_key: TiNAGG4BaaMdaH1tRfuU:KnR6yE41RrSowb0kQ0HWoA <1>
 --------------------
-<1> Format is `id:api_key` (as returned by {ref}/security-api-create-api-key.html[Create API key])
+<1> Format is `id:api_key` (as shown in the Beats dropdown)
 
 [float]
 [[beats-api-key-monitor]]
 === Create an API key for monitoring
-To create an API key to use for sending monitoring data to {es}, use the
-{ref}/security-api-create-api-key.html[Create API key API], for example:
+
+In {kib}, navigate to **Stack Management** > **API keys** and click **Create API key**.
+
+[role="screenshot"]
+image::images/server-api-key-create.png[API key creation]
+
+Enter a name for your API key and select **Restrict privileges**.
+In the role descriptors box, assign the appropriate privileges to the new API key.
+For example:
+
+[source,json,subs="attributes,callouts"]
+----
+{
+    "{beat_default_index_prefix}_monitoring": {
+        "cluster": ["monitor"], <1>
+        "index": [ <2>
+            {
+                "names": [".monitoring-beats-*"],
+                "privileges": ["create_index", "create"]
+            }
```

Comment:
Same question here. What we had is different from what we show in [**Grant privileges and roles needed for monitoring**](https://www.elastic.co/guide/en/apm/server/current/privileges-to-publish-monitoring.html).

Reply:
Same as above, let's rely on the privilege docs.